cpp/src/arrow/compute/exec/aggregate_node.cc

```diff
@@ -352,15 +493,34 @@ class GroupByNode : public ExecNode {
       int key_field_id = key_field_ids[i];
       output_fields[base + i] = input_schema->field(key_field_id);
     }
+    base += keys.size();
+    for (size_t i = 0; i < segment_keys.size(); ++i) {
+      int segment_key_field_id = segment_key_field_ids[i];
+      output_fields[base + i] = input_schema->field(segment_key_field_id);
```

Comment:
I don't think `output_fields` is large enough to do `output_fields[base + i]`.  I get a segmentation fault here.

Are the segment keys part of the output?  I would expect them to be. However, I also don't see any place in GroupByNode::Finalize that is putting the segment keys into the output data.